configure.ac

```diff
@@ -376,12 +376,50 @@ fi
 
 #fi
 
-# What SVN revision are we using?
-define([svnversion], esyscmd([sh -c "svnversion -n"]))dnl
-if test "x$want_cuda" != "xUnversioned directory"; then
- AC_DEFINE(CISTEM_SVN_REV, "svnversion", [SVN Revision])
+# What SVN revision are we using? Inner if clause will not work if svnversion is not callable
+AC_CHECK_PROG(SVB_CHECK,svnversion,yes)
+if test "x$SVN_CHECK" == "xyes" ; then
+  define([svnversion], esyscmd([sh -c "svnversion -n"]))dnl
+  if test "x$svnversion" != "xUnversioned directory"; then
+  AC_DEFINE(CISTEM_SVN_REV, "svnversion", [SVN Revision])
+  fi
 fi
 
+if test x$compile_experimental = xtrue ; then
+
+AC_CHECK_PROG(GIT_CHECK,git,yes)
+AC_CHECK_FILE(/groups/himesb/git/cisTEM_downstream_bah/.git/HEAD,
+  [
+    if test "x$GIT_CHECK" == "xyes"; then 
+      # Get the current branch, which could have multiple refs, which is why we need the awk (*) marks the active branch.
+      AC_SUBST([CISTEM_CURRENT_BRANCH],[`git  branch --points-at  HEAD | awk '/^\*/{print $2}'`])
+      # Get the first reachable tag - number of commits since that tag - short commit hash
+      AC_SUBST([CISTEM_VERSION_tag_nCommits_commitish],[`git describe --tags`])
```

Comment:
I quite liked the idea of adding the `--dirty` option here. Did you consider it? If you chose not to use it, I'm interested in your thinking.

Reply:
This is a good idea - knowing the commit may not help to trouble shoot if the working tree is not clean. I've added this.